hawkular-command-gateway-war/src/main/java/org/hawkular/cmdgw/ws/command/ForwardCommand.java

```diff
@@ -26,37 +23,48 @@
 import org.hawkular.bus.common.MessageId;
 import org.hawkular.bus.common.MessageProcessor;
 import org.hawkular.bus.common.producer.ProducerConnectionContext;
-import org.hawkular.cmdgw.api.DeployApplicationRequest;
+import org.hawkular.cmdgw.api.AuthMessage;
+import org.hawkular.cmdgw.api.AuthMessage.RecipientType;
 import org.hawkular.cmdgw.api.GenericSuccessResponse;
 import org.hawkular.cmdgw.ws.Constants;
 import org.hawkular.cmdgw.ws.MsgLogger;
-import org.hawkular.cmdgw.ws.command.Command;
-import org.hawkular.cmdgw.ws.command.CommandContext;
-import org.hawkular.inventory.api.model.CanonicalPath;
 
 /**
  * UI client requesting to send a file to a remote feed.
  */
-public class DeployApplicationCommand implements Command<DeployApplicationRequest, GenericSuccessResponse> {
-    public static final Class<DeployApplicationRequest> REQUEST_CLASS = DeployApplicationRequest.class;
+public class ForwardCommand implements Command<AuthMessage, GenericSuccessResponse> {
 
     @Override
-    public BasicMessageWithExtraData<GenericSuccessResponse> execute(DeployApplicationRequest request,
-            BinaryData binaryData, CommandContext context) throws Exception {
+    public BasicMessageWithExtraData<GenericSuccessResponse> execute(AuthMessage request, BinaryData binaryData,
+            CommandContext context) throws Exception {
 
-        // determine what feed needs to be sent the message
-        CanonicalPath resourcePath = CanonicalPath.fromString(request.getResourcePath());
-        String feedId = resourcePath.ids().getFeedId();
+        RecipientType recipientType = request.getRecipientType();
```

Comment:
Yes, ForwardToUIClient and ForwardToFeed could do the job, but I vote for keeping the gateway as simple (and message details agnostic) as possible. If there is a way to avoid analyzing `resourcePath` and thus a way to avoid dependending on inventory, I think that we should go that way.